subprojects/core-api/src/main/java/org/gradle/api/Project.java

```diff
@@ -427,6 +428,14 @@ public interface Project extends Comparable<Project>, ExtensionAware, PluginAwar
      */
     Project getProject();
 
+    /**
+     * <p>Returns an immutable view of this project, safe for use with isolated projects.</p>
+     *
+     * @return This project as an {@link ImmutableProject}. Never returns null.
+     */
+    @NotNull
+    ImmutableProject getAsImmutableProject();
```

Comment:
💭 We are strongly considering naming these `IsolatedProject` and `getIsolated()` respectively so the DSL pattern to access those views, when available, would be `model.isolated.property`, e.g., `project.isolated.rootDirectory`. Thoughts?

Reply:
I think that makes a lot of sense. Do you want me to update this PR?

Reply:
Yes, please, that would be great 👍

Reply:
Rebased onto latest master and pushed the name change as a new commit.